exchanges/huobi/huobi_test.go

```diff
@@ -1839,22 +1846,21 @@ func TestPairFromContractExpiryCode(t *testing.T) {
 			continue // Next Quarter is intermittently present
 		}
 		require.NoErrorf(t, err, "pairFromContractExpiryCode must not error for %s code", cType)
-		assert.Equal(t, currency.BTC, p.Base, "pair Base should be the same")
+		assert.Equal(t, currency.BTC, p.Base, "pair Base should be BTC")
 		h.futureContractCodesMutex.RLock()
-		exp, ok := h.futureContractCodes[cType]
+		cachedContract, ok := h.futureContractCodes[cType]
 		h.futureContractCodesMutex.RUnlock()
 		require.True(t, ok, "%s type must be in contractExpiryNames", cType)
-		assert.Equal(t, currency.BTC, p.Base, "pair Base should be the same")
-		assert.Equal(t, exp, p.Quote, "pair Quote should be the same")
-		d, err := time.ParseInLocation("060102", p.Quote.String(), tz)
+		assert.Equal(t, cachedContract, p.Quote, "pair Quote should match contractExpiryNames")
+		exp, err := time.ParseInLocation("060102", p.Quote.String(), tz)
 		require.NoError(t, err, "currency code must be a parsable date")
-		require.Falsef(t, d.Before(n), "%s expiry must be today or after", cType)
-		switch cType {
-		case "CW", "NW":
-			require.Truef(t, d.Before(n.AddDate(0, 0, 14)), "%s expiry must be within 14 days; Got: `%s`", cType, d)
-		case "CQ", "NQ":
-			require.Truef(t, d.Before(n.AddDate(0, 6, 0)), "%s expiry must be within 6 months; Got: `%s`", cType, d)
-		}
+		require.Falsef(t, exp.Before(today), "%s expiry must be today or after; Got: `%s`", cType, exp)
+		diff := uint(exp.Sub(today).Hours() / 24)
+		require.LessOrEqual(t, diff, expiryWindows[cType], "%s expiry must be within expected update window; Today: `%s`, Expiry: `%s`",
```

Comment:
Suggestion: Could use `InDelta` func up to you

Reply:
Makes my brain hurt slightly to think about using a Delta for this.
Would that be like using expiryWindows[cType] as the delta and 0 as the expected value?
I did consider using `WithinDuration` but decided this was clearer to parse in both code and test failure situations.

Reply:
That is a good point!